### PR TITLE
Update to pyglet 1.4.1, add freetype dependency, enable more tests, switch to using pip

### PR DESCRIPTION
**Pull request by hoechenberger**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR
- updates `pyglet` to 1.4.1 (makes #11 and #12 obsolete)
- adds the `freetype` run dependency (#9),
- enables more `import` tests
- switches the installation from using  `setuptools` to `pip`
- adds the previously missing `future` dependency

### CODE REVIEW

**Comment by hoechenberger** [comment]
Friendly ping to the maintainers :)

**Comment by hoechenberger** [comment]
I've rebased to hopefully make the review easier.

**Comment by hoechenberger** [comment]
The latest commit will update to recipe to build `pyglet` 1.4.0, making #11 obsolete.

**Comment by hoechenberger** [comment]
Friendly ping to the maintainers.

File: recipe/run_tests.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    # skip to avoid installing audio drivers on Windows, see
+    # https://www.openal.org/platforms/
+    # import pyglet.media.drivers.openal
+    # pyglet.media.drivers.pulse
+
+if sys.platform == 'darwin':
+    import pyglet.libs.darwin
+    import pyglet.libs.darwin.cocoapy
+    import pyglet.window.cocoa
+    # import pyglet.media.drivers.pulse
```
Comment (saraedum) [inline recipe/run_tests.py]: Is the reason for skipping the same as above? Then you should probably mention this here (and in the Linux section below.)

Comment (hoechenberger) [reply recipe/run_tests.py]: Thank you, I have addressed this in 05e5bf9

**Comment by mariusvniekerk** [comment]
@hoechenberger can add yourself as a maintainer here, please?

**Comment by hoechenberger** [comment]
@mariusvniekerk Done. 

The Python 3.7 Linux build failed on Azure needs to be restarted, it seems…

**Comment by jakirkham** [comment]
Restarted 🙂

**Comment by hoechenberger** [comment]
@jakirkham Thank you! All tests have passed now.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -12,60 +12,31 @@ source:
...
+  commands:
+    - python run_tests.py  # [not linux]
+    - xvfb-run --server-args=":99 -screen 0 1024x768x24 -ac +extension GLX -noreset" python run_tests.py  # [linux]
 
 about:
-  home: http://pyglet.readthedocs.org/en/pyglet-1.2-maintenance/
-  license: BSD 3-Clause
+  home: https://pyglet.readthedocs.io/en/pyglet-1.3-maintenance/
+  license: BSD-3-Clause
+  license_family: BSD
```
Comment (jakirkham) [inline recipe/meta.yaml]: Would be good to package the license file as well like so...

```suggestion
  license_family: BSD
  license_file: LICENSE
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -12,60 +12,31 @@ source:
...
-    - pyglet.graphics  # [win]
-    - pyglet.image  # [win]
-    - pyglet.image.codecs  # [win]
-    - pyglet.input  # [win]
-    - pyglet.media  # [win]
-    - pyglet.media.drivers  # [win]
-    - pyglet.media.drivers.directsound  # [win]
-    - pyglet.window  # [win]
-    - pyglet.text  # [win]
-    - pyglet.text.formats  # [win]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Why are we removing the import tests?

Comment (hoechenberger) [reply recipe/meta.yaml]: @jakirkham 
Some tests require access to a (virtual) screen on Linux; therefore, the tests need to be run through `xvfb-run` on that platform. To do that, I created a Python script, `run_tests.py`, that does all the `import` testing. It is run directly (`python run_tests.py`) on Windows and macOS, and via `xvfb-run` on Linux. I have added a comment explaining this in 6e1dc47

An alternative could be to only list the (few) tests that actually require a virtual screen in the test script, and keeping the rest of the `import tests` in `meta.yaml`. But that would put the tests in two different places, and could be confusing?

**Comment by hoechenberger** [comment]
@mariusvniekerk Thank you!

**Pull request merged**